### ISSUE TITLE
Improve record deletion options

### ISSUE THREAD

**Issue by jacobthill** [sul-dlss/dlme#527]
Reminder: currently there is no way for @jacobthill to delete metadata from stage or prod once the url to the S3 bucket is no longer visible in the admin interface (the interface only displays the 20 most recent transformations). 

![transform-results](https://user-images.githubusercontent.com/37662787/61798481-5f5d7b00-adf7-11e9-91db-1bfb6e97e069.png)

A more user friendly solution would add the ability to delete records associated with a given data, instead of deleting records by aws url (e.g. penn/egyptian would look for every aws url that resulted from that path and delete them one-by-one starting with the most recent and continuing until there are no more penn/egyptian records left in the application).

**Comment by jacobthill** [comment]
Also, some collections, for unknown reasons, cause errors when trying to delete records. For example, the [Palestine Poster Project Archive](https://spotlight.stage.dlmenetwork.org/library/catalog?f%5Bsource_institution%5D%5B%5D=Palestine+Poster+Project+Archive) I added these records: https://s3-us-west-2.amazonaws.com/dlme-metadata-stage/output-d3fad795-0404-4518-99ef-0d798e98c218.ndjson

then updated the config file to include a rights statement and added them again:
https://s3-us-west-2.amazonaws.com/dlme-metadata-stage/output-6ac53307-ba38-47ac-a2dd-ee4c69031616.ndjson

 but they didn't update (e.g. the rights statement wasn't included which was likely because I mapped it to cho_rights instead of cho_dc_rights), so I used both urls to delete the items. All of the records were deleted except 3. We need to figure out why this happened and how to prevent it from happening again.

**Comment by jacobthill** [comment]
Also, some collections, for unknown reasons, cause errors when trying to delete records. For example, the Palestine Poster Project Archive I added these records: https://s3-us-west-2.amazonaws.com/dlme-metadata-stage/output-d3fad795-0404-4518-99ef-0d798e98c218.ndjson

then updated the config file to include a rights statement and added them again:
https://s3-us-west-2.amazonaws.com/dlme-metadata-stage/output-6ac53307-ba38-47ac-a2dd-ee4c69031616.ndjson

but they didn't update (e.g. the rights statement wasn't included which was likely because I mapped it to cho_rights instead of cho_dc_rights), so I used both urls to delete the items. All of the records were deleted except 3. We need to figure out why this happened and how to prevent it from happening again.
